### PR TITLE
use front-matter in index.html files

### LINKED ISSUE #643: T - use front-matter in the index.html instead of a index.yaml file
At the moment, the dumper creates two files:

* `index.html` - just the HTML
* `index.yaml` - just the metadata in YAML

Instead, we should just have 1:

* `index.html` - metadata in YAML, as front-matter, above the HTML

This affects...

* `processFolder`
* `processFolderTitle`
* the `index.hash` caching
* and those various things like "Edit this page in your editor"
* the macros flaws links need to point to a different line number

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #643

To test:
1. `node content import --start-clean`
2. Check the generated files in `content/files/**/index.html`. No `index.yaml` file!
3. `yarn clean && yarn start`
4. Visit a page that has flaws. E.g. http://localhost:3000/en-US/docs/Web/CSS/background-position#_flaws
5. Check that clicking the lines in the flaws opens the right spot. 
6. Use the in-browser editor http://localhost:3000/en-US/_edit/Web/CSS/background-position and it should simply work. 

When you run `yarn start` you'll see something like this:
```
5:29:02 PM watchcontent.1 |  File watcher set up for /Users/peterbe/dev/MOZILLA/MDN/yari/content/files. Watching over 44,978 files in 44,960 folders.
```
Which is great because it's half as many files to watch.

### CODE REVIEW

**Comment by peterbe** [comment]
@escattone Latest master with all the live sample fixes have now been merged into this. 

We really need eslint! There were at least two bugs that I only found out about after having run the `node content build ...` after a long time. It might be an effect of my manual git merge conflict resolution. But it's still sad that we're suffering when a computer could have done the work for us.

**Comment by peterbe** [comment]
Note-to-self; This isn't making the right link to GitHub ("Edit this page on GitHub").

File: content/scripts/document.js [hunk 1]
```diff
@@ -7,23 +7,23 @@ function buildPath(contentPath, slug) {
...
+    fs.writeFileSync(
       path.join(folder, "wikihistory.json"),
       JSON.stringify(wikiHistory, null, 2)
     );
   }
 }
 
-module.exports = { buildPath, create };
+function saveFile(filePath, html, metadata) {
+  const combined = `---\n${yaml.safeDump(metadata)}\n---\n${html.trim()}\n`;
```
Comment (escattone) [inline content/scripts/document.js:25]: Nit. There's an extra line break after the `yaml.safeDump()`, for example:
```yaml
---
title: Game development
slug: Games
summary: >-
  Gaming is one of the most popular computer activities. New technologies are
  constantly arriving to make it possible to develop better and more powerful
  games that can be run in any standards-compliant web browser.
tags:
  - Apps
  - Game Development
  - Gamedev
  - Games
  - HTML5 Games
  - JavaScript Games
  - Web

---
```
```suggestion
  const combined = `---\n${yaml.safeDump(metadata)}---\n${html.trim()}\n`;
```

File: content/scripts/build.js [hunk 2]
```diff
@@ -222,17 +222,23 @@ function extractLocale(source, folder) {
...
 
-function getMetadata(source, folder) {
-  const metadataRaw = fs.readFileSync(path.join(folder, "index.yaml"));
-  const metadata = yaml.safeLoad(metadataRaw);
-  const wikiHistoryPath = path.join(folder, "wikihistory.json");
-  if (fs.existsSync(wikiHistoryPath)) {
-    const wikiMetadataRaw = fs.readFileSync(wikiHistoryPath);
-    const wikiMetadata = yaml.safeLoad(wikiMetadataRaw);
-    metadata.modified = wikiMetadata.modified;
+function getContent(source, folder, appendWikiHistory = false) {
```
Comment (escattone) [inline content/scripts/build.js:225]: ❤️  the `appendWikiHistory = false`, since most of the time we don't need the `modified` attribute. 👍

**Pull request merged**